### PR TITLE
improvement(ViT): use Crop to subtitude Gather

### PR DESCRIPTION
**Pull request by tpoisonooo**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily receiving feedbacks. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Fix  review advices from https://github.com/Tencent/ncnn/pull/3730

## Modification

* mmdeploy split nn.Linear into multiple opr when input.shape=[1, x, x]
  * remove ~100 redundant opr in `end2end.param` (289 --> 195)
* Use `Crop axis=0 starts=0 ends=1` opr instead of  `Gather axis=0 indices=[0]` in ViT.param
* Add linear testcase
* Enable fast GELU by default (from nihui)

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit tests to ensure the correctness.
3. If the modification has a dependency on downstream projects of a newer version, this PR should be tested with all supported versions of downstream projects.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

**Comment by tpoisonooo** [comment]
@grimoire  review advices all accepted.

File: mmdeploy/pytorch/ops/linear.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        bias,
+        alpha_f=1.0,
+        beta_f=1.0,
+        transA_i=0,
+        transB_i=1)
+
+
+@SYMBOLIC_REWRITER.register_symbolic(
+    'linear', is_pytorch=True, backend=Backend.NCNN.value)
+def linear__ncnn(ctx, g, input, weight, bias):
```
Comment (grimoire) [inline mmdeploy/pytorch/ops/linear.py:39]: It is surprising that there is no symbolic function for linear before torch1.9 ...
If we want to support torch<1.9, we might need to add a pass to fuse the ops. Or should we require torch>=1.9?

Comment (tpoisonooo) [reply mmdeploy/pytorch/ops/linear.py:39]: If rewrite `linear` not matched, the ncnn model still works.

Old version is burden, add torch version requirement would be easy +_+

**Review by grimoire — DISMISSED** [review]
LGTM

File: tests/test_pytorch/test_pytorch_ops.py [hunk 2]
```diff
@@ -127,6 +127,27 @@ def test_instance_norm():
...
+class TestLinear:
+
+    def test_normal(self):
+        x = torch.rand(1, 2, 3)
+        w = torch.rand(2, 3)
+        bias = torch.rand(2)
+        model = OpModel(torch.nn.functional.linear, w, bias).eval()
+        nodes = get_model_onnx_nodes(model, x)
+        print(nodes)
+        assert nodes[0].op_type == 'Matmal'
```
Comment (RunningLeon) [inline tests/test_pytorch/test_pytorch_ops.py:140]: `Matmal` ?

File: tests/test_pytorch/test_pytorch_ops.py [hunk 3]
```diff
@@ -127,6 +127,27 @@ def test_instance_norm():
...
+
+    def test_no_bias(self):
```
Comment (RunningLeon) [inline tests/test_pytorch/test_pytorch_ops.py:142]:
```
___________________________________________ TestLinear.test_no_bias ____________________________________________

self = <test_pytorch_ops.TestLinear object at 0x7f977323a510>

    def test_no_bias(self):
        x = torch.rand(1, 2, 3)
        w = torch.rand(2, 3)
        model = OpModel(torch.nn.functional.linear, w).eval()
        nodes = get_model_onnx_nodes(model, x)
        print(nodes)
>       assert nodes[0].op_type == 'MatMul'
E       AssertionError: assert 'Constant' == 'MatMul'
E         - MatMul
E         + Constant

tests/test_pytorch/test_pytorch_ops.py:148: AssertionError
```

File: mmdeploy/pytorch/ops/linear.py [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+# Modified from:
+# https://github.com/pytorch/pytorch/blob/9ade03959392e5a90b74261012de1d806cab2253/torch/onnx/symbolic_opset9.py
+from torch.onnx.symbolic_helper import parse_args
+
+from mmdeploy.core import SYMBOLIC_REWRITER
+from mmdeploy.utils import Backend
+
+
+@parse_args('v', 'v', 'f', 'f', 'i', 'i')
+def linear_no_bias(g, input, weight):
```
Comment (RunningLeon) [inline mmdeploy/pytorch/ops/linear.py:11]: Seems to be similliar function as below, should we keep both?
https://github.com/open-mmlab/mmdeploy/blob/32482e76fde68839890ff184cb2d2adb5de11879/mmdeploy/pytorch/functions/linear.py#L0-L51

Comment (tpoisonooo) [reply mmdeploy/pytorch/ops/linear.py:11]: Yes keep both. As I understand it, FUNCTION_REWRITER will work and then OPS_REWRITER will take effect.

File: tests/test_pytorch/test_pytorch_ops.py (hunk 2 above)
Comment (tpoisonooo) [reply tests/test_pytorch/test_pytorch_ops.py]: Done.

File: tests/test_pytorch/test_pytorch_ops.py (hunk 3 above)
Comment (tpoisonooo) [reply tests/test_pytorch/test_pytorch_ops.py:142]: Done.  And this is my output:
```
------------------------------------------------------------------------- Captured stdout call --------------------------------------------------------------------------
[input: "onnx::MatMul_0"
input: "onnx::MatMul_4"
output: "3"
name: "MatMul_0"
op_type: "MatMul"
]
```

Comment (RunningLeon) [reply tests/test_pytorch/test_pytorch_ops.py:142]: @tpoisonooo 
still fails on my side: pytorch==1.8.0.
the index of node is not right.
```
=============================================================== FAILURES ================================================================
________________________________________________________ TestLinear.test_normal _________________________________________________________

self = <test_pytorch_ops.TestLinear object at 0x7faebef27a50>

    def test_normal(self):
        x = torch.rand(1, 2, 3)
        w = torch.rand(2, 3)
        bias = torch.rand(2)
        model = OpModel(torch.nn.functional.linear, w, bias).eval()
        nodes = get_model_onnx_nodes(model, x)
        print(nodes)
>       assert nodes[0].op_type == 'MatMul'
E       AssertionError: assert 'Constant' == 'MatMul'
E         - MatMul
E         + Constant

tests/test_pytorch/test_pytorch_ops.py:140: AssertionError
--------------------------------------------------------- Captured stdout call ----------------------------------------------------------
[output: "1"
name: "Constant_0"
op_type: "Constant"
attribute {
  name: "value"
  t {
    dims: 3
    dims: 2
    data_type: 1
    raw_data: "\370\271\215>\236(\326>u\2321?\333\3177?\203\026\026?\260\036\262="
  }
  type: TENSOR
}
, input: "input"
input: "1"
output: "2"
name: "MatMul_1"
op_type: "MatMul"
, output: "3"
name: "Constant_2"
op_type: "Constant"
attribute {
  name: "value"
  t {
    dims: 2
    data_type: 1
    raw_data: "\344U\230> z\237="
  }
  type: TENSOR
}
, input: "2"
input: "3"
output: "4"
name: "Add_3"
op_type: "Add"
]
________________________________________________________ TestLinear.test_no_bias ________________________________________________________

self = <test_pytorch_ops.TestLinear object at 0x7faebef27d50>

    def test_no_bias(self):
        x = torch.rand(1, 2, 3)
        w = torch.rand(2, 3)
        model = OpModel(torch.nn.functional.linear, w).eval()
        nodes = get_model_onnx_nodes(model, x)
        print(nodes)
>       assert nodes[0].op_type == 'MatMul'
E       AssertionError: assert 'Constant' == 'MatMul'
E         - MatMul
E         + Constant

tests/test_pytorch/test_pytorch_ops.py:148: AssertionError
--------------------------------------------------------- Captured stdout call ----------------------------------------------------------
[output: "1"
name: "Constant_0"
op_type: "Constant"
attribute {
  name: "value"
  t {
    dims: 3
    dims: 2
    data_type: 1
    raw_data: "\267by?B{\351>\240\371\354<\207\214Y?\376\3436?%\n\033?"
  }
  type: TENSOR
}
, input: "input"
input: "1"
output: "2"
name: "MatMul_1"
op_type: "MatMul"
]


```

**Review by grimoire — DISMISSED** [review]

**Review by RunningLeon — APPROVED** [review]
LGTM

**Review by grimoire — APPROVED** [review]

**Pull request merged**